src/client/opamArg.ml

```diff
@@ -697,6 +700,10 @@ let apply_build_options cli b =
     ?fake:(flag b.fake)
     ?skip_dev_update:(flag b.skip_update)
     ?assume_depexts:(flag (b.assume_depexts || b.no_depexts))
+    ?verbose_on:
+      (b.verbose_on >>= function
+        | [] -> None
+        | vo -> Some (OpamPackage.Name.Set.of_list vo))
```

Comment:
Current behaviour: is flag list is empty, it is considered as nothing was given. Same for environment variable.